### DISCUSSION TITLE
Stopping the Fuzzing without using Keyboard interrupt

### DISCUSSION THREAD

**Discussion by ACBNair** [jtpereyda/boofuzz#600] [Q&A] [answered]
Is it possible to stop the test without killing the application (keyboard interrupt)? I have a scenario that if a certain condition mets, I need to stop the testing.

**Comment by SR4ven** [comment, marked as answer]
Two options come to my mind:
1. Raise any exception which is not caught in boofuzz and catch it in your script calling `session.fuzz`.
2. If you have access to the session object, set `session._index_end = 0`. That will break the main fuzzing loop.

You might also want to set `keep_web_open=False` in `Session` so that boofuzz will not block holding the website open.

**Reply by ACBNair** [reply]
Thank you @SR4ven . 

 `_index_end = 0` is helpful. Is it advisable to do this? There can be scenarios where we need to stop the test. Do we plan a method for that in boofuzz?

We are not using in web mode. Though what is the impact of `keep_web_open`?

**Reply by SR4ven** [reply]
Depends on how gracefully you want to stop the testing. Setting `_index_end = 0` should be quite graceful. Raising a custom exception that you catch in your own script might be a bit faster but will end boofuzz more abruptly.

Currently there is no feature planned to stop fuzzing as far as I know. I guess it would also depend on where you want to stop fuzzing from e.g. monitor/callback. However, if you come up with neat solution, fell free to open a PR or just write a feature request.

`keep_web_open` will make session.fuzz block after the fuzzing is complete so that the user can check the results on the webinterface on port 26000. If you don't need the webinterface you can disable it with the session arg `web_port = None`. `keep_web_open` will not have any effect in this case.
